pennylane_lightning/core/src/simulators/lightning_tensor/tncuda/observables/ObservablesTNCudaOperator.hpp

```diff
@@ -168,84 +260,229 @@ template <class TensorNetT> class ObservableTNCudaOperator {
 
   public:
     ObservableTNCudaOperator(const TensorNetT &tensor_network,
```

Comment:
I suggest simplifying the constructor. It became a complex method that takes care of all obs and corner cases. You can create helper "private" methods for each obs / cases.

Reply:
That or overloading might help.